### PR TITLE
Create rolebinding for .Release.Namespace implicitly

### LINKED ISSUE #641: `Restricted` (minimal) RBAC permission cannot `watch` and `list` secrets in Keda namespace in Keda-operator
Hi there,

We are currently using the Keda chart `v2.13.2` and everything works perfectly. After we upgraded to `v2.14.2` we are getting the following error in the component `Keda-operator`:
```
E0521 18:16:46.425051       1 reflector.go:147] k8s.io/client-go/informers/factory.go:150: Failed to watch *v1.Secret: failed to list *v1.Secret: secrets is forbidden: User "system:serviceaccount:keda:keda-operator" cannot list resource "secrets" in API group "" in the namespace "keda"
W0521 18:16:50.015001       1 reflector.go:535] k8s.io/client-go/informers/factory.go:150: failed to list *v1.Secret: secrets is forbidden: User "system:serviceaccount:keda:keda-operator" cannot list resource "secrets" in API group "" in the namespace "keda"
```

## Expected Behavior

The Keda-operator would be up and running.

## Actual Behavior

The Keda-operator is crashing.

## Steps to Reproduce the Problem

  1. Using `v2.14.2` of the helm chart
  2. Put the following in the values:
  ```
  permissions:
    metricServer:
      restrict:
        secret: true
    operator:
      restrict:
        secret: true
  ```

## Specifications

  - **KEDA Version:** v2.14 (Keda helm chart v2.14.2)
  - **Platform & Version:** EKS v1.29
  - **Kubernetes Version:** v1.29

### PR DESCRIPTION
**Pull request by jkremser**
When helm chart is installed with a non-empty `watchNamespace` field, it will create the `RoleBinding` resource for the cluster role that contains most of the operator rights for each specified namespace in this CSV field. However, not for the namespace into which we actually install KEDA (mostly called `keda`).

So this PR adds the `.Release.Namespace` to the list

### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] README is updated with new configuration values *(if applicable)* [learn more](https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#documentation)
- [x] A PR is opened to update KEDA core ([repo](https://github.com/kedacore/keda)) *(if applicable, ie. when deployment manifests are modified)* N/A

Fixes #641

### CODE REVIEW

**Review by zroubalik — APPROVED** [review]
LGTM, thanks for the fix

**Comment by kamialie** [comment]
I guess you could also increment patch part of the Helm chart version.

**Comment by calvinbui** [comment]
i dont think this will solve the problem as the `clusterrole` is still missing the `list` and `watch` actions if `.Values.permissions.operator.restrict.secret: true`?

https://github.com/kedacore/charts/blob/v2.14.2/keda/templates/manager/clusterrole.yaml#L33-L40

File: keda/templates/manager/minimal-rbac.yaml [hunk 1]
```diff
@@ -34,6 +34,10 @@ rules:
   verbs:
   - create
   - update
+{{- if .Values.permissions.operator.restrict.secret }}
+  - list
+  - watch
+{{- end }}
```
Comment (jkremser) [inline keda/templates/manager/minimal-rbac.yaml:40]: this is little bit counter intuitive, but this means that if the secret access should be restricted, it's restricted [here](https://github.com/kedacore/charts/blob/6aa60122e1dbd7d2ccf0a904eda677a01ae9fc05/keda/templates/manager/clusterrole.yaml#L33-L50), but the operator needs list & watch at least for the `.Release.Namespace` (defaults to `keda`)

**Review by zroubalik — APPROVED** [review]

**Review by JorTurFer — APPROVED** [review]

**Comment by joebowbeer** [comment]
How is the documentation regarding [restricting access to secrets](https://keda.sh/docs/2.15/operate/cluster/#restrict-secret-access) affected by #625 and this change?

https://github.com/kedacore/charts/issues/685#issuecomment-2342545761

**Pull request merged**